### ISSUE TITLE
Docker Hub new policy means nightlies / pull-request-checks will start to fail 1st of November

### ISSUE THREAD

**Issue by TrueBrain** [OpenTTD/OpenTTD#8309]
OpenTTD's Continues Integration uses Docker Hub to pull the images to validate linux builds and to create linux binaries:

https://hub.docker.com/r/openttd/compile-farm
https://hub.docker.com/r/openttd/compile-farm-ci

On the 1st of November a new Docker Hub policy will be in place, which restricts the amount of downloads you can do from Docker Hub if you are not authorized and/or if you do not pay. (https://www.docker.com/pricing, "Data transfer"; see https://www.docker.com/pricing/resource-consumption-updates for more details what a "pull" is).
Without action, it is -very- likely OpenTTD will fail to produce nightlies from the 1st of November, and Pull Requests will fail their checks. (of course, it has to be seen what happens exactly as we use Azure DevOps to do the building, which constantly has different IPs.)

There are, as far as I am aware, three directions to look for a solution:

1) migrate Docker Hub to GitHub Container Registry. This is relative painless, and only requires changing the URL for images that are pushed and pulled.
2) pay Docker Hub 5 euro a month, and start using an authenticated account to do the "docker pull". Also requires minor changes, but only on the pull side of things. (the push side already uses the OpenTTD account.)
3) switch to GitHub Actions, and integrate the CompileFarm repository (https://github.com/OpenTTD/CompileFarm) in there, creating the docker just before it is being used. This is a bit wasteful in CPU-cycles, but in general the way GitHub Actions are currently being used by the community as a whole.

Either way, to be sure there is no wrong assumption here, I (TrueBrain) will **-not-** pick up this ticket. Of course you are always free to pick my brain on IRC or via this ticket. https://github.com/OpenTTD/actions/commit/f005561eb930a8f7aa411441024cb47db6748509 might give some inspiration how I resolved this issue for the OpenTTD infrastructure on AWS.

In summary, without any action from any developers, on the first of november two things will (most likely) stop working:

- Pull Requests "checks"
- Nightlies (and extending on that, new releases)

**Comment by orudge** [comment]
From a financial point of view, option 2 is no problem, so if we're not able to get one of the other options to work, then I'm happy for us to pay the relatively small fee for Docker Hub.

Option 1 might be the simplest to try to implement in the short term at least.
